### ISSUE TITLE
lesson2-convbattle2-linear-model-with-CNN-features-Copy1

### ISSUE THREAD

**Issue by vincentporte** [IALeMans/Fast.ai-17#4]
whatever learning rate, accuracy and validation accuracy sticks to original values.

to check : 
- train_data.bc & valid_data.bc contents
- val_classes, trn_classes, vall_labels & trn_labels 
- batches = gen.flow(trn_data, trn_labels, batch_size=batch_size, shuffle=True)
- val_batches = gen.flow(val_data, val_labels, batch_size=batch_size, shuffle=False)
- test with catsdogs dataset

@AMairesse @jbcotard @ithayiga

**Comment by vincentporte** [comment]
il existe 2 méthodes pour generer les batches d'images qui alimentent l'entrainement : 

- ImageDataGenerator.flow qui utilisent des tableaux de données et de labels, ceux générés par la fonction get_data
- ImageDataGenerator.flow_from_Directory qui récupèrent directement les images dans les répertoires

Les deux versions de batches sont envoyées en entrée du model.fit_generator mais avec un syntaxe légèrement différente. Peut-être à l'origine du défaut rencontrés ?

**Comment by vincentporte** [comment]
Le probleme semblait venir du jeu de données. En réutilisant convbattle2 dans un notebook fonctionnel, le résultat était KO.
Finalement, certainement un probleme dans le jeu d'image préprocessées [fast.ai/convbattle2/models/train_data.bc et valid_data.bc]. Apres regeneration correcte, et skipping de l'étape dans les notebooks suivant, tout est ok.

**Comment by vincentporte** [comment]
peut être un probleme de minimum local dans le jeu de données.
fonctionne régulièrement bien, quelquefois non, avec code similaire

**Comment by vincentporte** [comment]
Hi, pour clore ce sujet, 2 options : 
1. soit un defaut dans le jeu de données (minimum local ou plateau)
1. un learning rate 10 fois trop grand quand l'accuracy sur le set de validation plafonne.
